expertsystem/data.py

```diff
@@ -126,3 +129,36 @@ def keys(self) -> KeysView[str]:
 
     def values(self) -> ValuesView[Particle]:
         return self.__particles.values()
+
+
+class QuantumState(NamedTuple):
+    """Collection of properties defining a quantum mechanical state.
+
+    Related to `.Particle`, but can contain more specific quantum state
+    information, such as `.Spin` projection.
+    """
+
+    charge: int
+    spin: Spin
+    mass: float
+    strangeness: int = 0
+    charmness: int = 0
+    bottomness: int = 0
+    topness: int = 0
+    baryon_number: int = 0
+    electron_number: int = 0
+    muon_number: int = 0
+    tau_number: int = 0
+    width: Optional[float] = None
+    isospin: Optional[Spin] = None
+    parity: Optional[Parity] = None
+    c_parity: Optional[Parity] = None
+    g_parity: Optional[Parity] = None
+
+
+class ComplexEnergyState(NamedTuple):
+    """Pole in the complex energy plane with quantum numbers."""
+
+    position: float
+    width: float
```

Comment:
To correlate better with the class name, `energy_real` and `energy_imaginary` would be better names in my opinion.

Reply:
What about using Python complex numbers (e.g. [this](https://docs.python.org/3/library/cmath.html))

Reply:
Yeah thats a good idea